services/s3/src/test/java/software/amazon/awssdk/services/s3/internal/crt/S3CrtAsyncHttpClientTest.java

```diff
@@ -509,6 +509,25 @@ void build_partSizeConfigured_shouldApplyToThreshold() {
         }
     }
 
+    @Test
+    public void build_noPartSize_shouldUseDefaultsForThresholdAndReadWindowSize() {
+        S3NativeClientConfiguration configuration =
+            S3NativeClientConfiguration.builder()
+                                       .credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create("test",
+                                                                                                                        "test")))
+                                       .signingRegion("us-west-2")
+                                       .build();
+        try (S3CrtAsyncHttpClient client =
+                 (S3CrtAsyncHttpClient) S3CrtAsyncHttpClient.builder()
+                                                            .s3ClientConfiguration(configuration).build()) {
+            long defaultPartSizeInBytes = 1024 * 1024L * 8L;
+            S3ClientOptions clientOptions = client.s3ClientOptions();
+            assertThat(clientOptions.getPartSize()).isEqualTo(0);
```

Comment:
`clientOptions.getPartSize() ==0 ` Is this intended ? 
- What happens if User explicitly sets partSize as as Zero Java SDK builder ?